### PR TITLE
First pass at documenting html5lib.filters

### PR DESCRIPTION
**Pull request by willkg**
This covers `html5lib.filters` in issue #307.

Is this documentation for filters clear enough for a 1.0?

### CODE REVIEW

**Comment by willkg** [comment]
@gsnedders, @jgraham Can one of you review this, please?

File: html5lib/filters/whitespace.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 
 class Filter(base.Filter):
-
+    """Collapses whitespace except in pre and textarea tags"""
```
Comment (gsnedders) [inline html5lib/filters/whitespace.py]: Elements, not tags. Also it's more than just pre and textarea; I think script is in rcdataElements too.

**Comment by willkg** [comment]
^^^ @gsnedders Does that fix the issue? I sort of understand it, but I'm not sure I know enough to know if it's completely fixed or not.

**Comment by willkg** [comment]
@gsnedders Can you review this please?

**Comment by willkg** [comment]
Thank you!

**Pull request merged**